pkg/timesync/timesync.go

```diff
@@ -54,7 +54,7 @@ const (
 
 // Default is the default timesync instance used throughout
 // the agent.
-var Default = NewTimeSync(context.Background(), "pool.ntp.org", 0)
```

Comment:
I just noticed this is not going to work. This is because `Default` will be defined before the configuration has been loaded, hence the NTPServer value will be empty.

```suggestion
var Default = NewTimeSync(context.Background(), "pool.ntp.org", 0)

// SetDefault overrides Default TimeSync instance
func SetDefault(ts *TimeSync) {
       Default = ts
}
```

Then on `main.go` you can override the default immediately after `LoadAgentConfig()` is called as follows:

```go
timesync.SetDefault(timesync.NewTimeSync(ctx, global.AgentConf.Runtime.NT
PServer, 0))
```